### PR TITLE
hw12

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/com/api/Comment.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.api;
+
+import com.google.api.client.util.DateTime;
+
+public class Comment {
```
Comment (Anderwerty) [inline src/main/java/com/api/Comment.java:5]: Why not use Lombok annotation here. Also, you could create a builder and use it.

File: src/main/java/com/api/YouTubeParser.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private static YouTube youtube;
+    private static List<String> scopes = Lists.newArrayList("https://www.googleapis.com/auth/youtube.force-ssl");
+    private List<YouTubeVideo> listOfVideos = Lists.newArrayList();
+
+    public void init() {
+        Credential credential = null;
+        try {
+            credential = Authorization.authorize(scopes, "commentThreads");
+        } catch (Exception e) {
+            LOGGER.info(e.getMessage());
```
Comment (Anderwerty) [inline src/main/java/com/api/YouTubeParser.java]: why info? why not error or warn?

File: src/main/java/com/api/YouTubeParser.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    public void getCommentsFromVideos(List<String> videoId) {
+        for (String video : videoId) {
+            List<CommentThread> videoComments = getCommentThreads(video);
+            CommentSnippet snippet;
+            YouTubeVideo youtubeVideo = new YouTubeVideo();
+
+            for (CommentThread videoComment : videoComments) {
+                snippet = videoComment.getSnippet().getTopLevelComment().getSnippet();
+                Comment comment = new Comment(snippet.getAuthorDisplayName(),
```
Comment (Anderwerty) [inline src/main/java/com/api/YouTubeParser.java]: Builder will better to use here.

File: src/main/java/com/api/YouTubeParser.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+
+    private static List<CommentThread> getCommentThreads(String videoId) {
+        CommentThreadListResponse videoCommentsListResponse = null;
+        try {
+            videoCommentsListResponse = youtube.commentThreads().list("snippet")
+                    .setVideoId(videoId).setTextFormat("plainText").execute();
+        } catch (IOException e) {
+            LOGGER.info(e.getMessage());
```
Comment (Anderwerty) [inline src/main/java/com/api/YouTubeParser.java]: The same.

File: src/main/java/com/api/YouTubeParser.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        return videoCommentsListResponse.getItems();
```
Comment (Anderwerty) [inline src/main/java/com/api/YouTubeParser.java]: In the case of IOException you will get NPE.

File: src/main/resources/client_secrets.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{"web":{
```
Comment (Anderwerty) [inline src/main/resources/client_secrets.json:1]: It better not to publish credential to the public repo.

**Review by Anderwerty — CHANGES_REQUESTED** [review]

File: src/main/java/com/api/Comment.java (hunk 1 above)
Comment (Sofasmile) [reply src/main/java/com/api/Comment.java:5]: May I use Lombok annotation next time, please?

File: src/main/java/com/api/YouTubeParser.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        try {
+            videoCommentsListResponse = youtube.commentThreads().list("snippet")
+                    .setVideoId(videoId).setTextFormat("plainText").execute();
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage());
+        }
+        if (videoCommentsListResponse != null) {
+            return videoCommentsListResponse.getItems();
+        }
+        throw new NullPointerException("NullPointerException");
```
Comment (Anderwerty) [inline src/main/java/com/api/YouTubeParser.java]: It is better to throw a custom exception with a reasonable description.

**Review by Anderwerty — APPROVED** [review]

**Pull request merged**